src/coins.cpp

```diff
@@ -199,7 +205,7 @@ bool CCoinsViewCache::BatchWrite(CCoinsMap &mapCoins, const uint256 &hashBlockIn
             } else {
                 // A normal modification.
                 cachedCoinsUsage -= itUs->second.coin.DynamicMemoryUsage();
-                itUs->second.coin = std::move(it->second.coin);
+                itUs->second.coin = erase ? std::move(it->second.coin) : it->second.coin;
```

Comment:
Should this also be updated to use an `if/else` block instead of a ternary like above?

Reply:
Good catch.